mozilla/fx-private-relay
# Add contacts model mpp 2259

This PR fixes MPP-2259. It builds on https://github.com/mozilla/fx-private-relay/pull/2303, so let's review & merge that first.
# New feature description
* Adds a new `InboundContact` model and API endpoints that will power the "Caller and Sender log" screen.
   * GET `/api/v1/inboundcontact/` shows all inbound contacts for the `RelayNumber` of the signed-in user
   * PATCH `/api/v1/inboundcontact/{id}` to set the `blocked:True|False` field of a contact
* Changes `inbound_sms` and `inbound_call` to:
   * Check if the profile has `store_phone_log:True|False`
   * If `True`, Get or create an `InboundContact` model for the caller/texter
   * If contact is blocked, increment the `num_calls_blocked` or `num_texts_blocked` field
   * If contact is allowed, increment the `num_calls` or `num_texts` field

# Screenshot (if applicable)
<img width="402" alt="image" src="https://user-images.githubusercontent.com/71928/183411787-2839b29d-4911-47a2-88e8-ca55743362e9.png">


How to test:
1. Make sure you've set up [end-to-end local phone development](https://github.com/mozilla/fx-private-relay/blob/main/docs/end-to-end-local-phone-dev.md)
2. Check out this branch and `python manage.py migrate`
3. Go to http://127.0.0.1:8000/api/v1/docs/
4. Run the `GET /api/v1/inboundcontact/` endpoint
   * [x] It should show 0 results
5. Send a text to your local Relay number
6. Run the `GET /api/v1/inboundcontact/` endpoint
   * [x] It should show 1 result
   * [x] Note the `last_inbound_date` value
7. Wait a little bit
8. Send a text to your local Relay number
9. Run the `GET /api/v1/inboundcontact/` endpoint
   * [x] It should show 1 result, and the `last_inbound_date` should be updated
10. Send a text to your local Relay number **FROM A DIFFERENT NUMBER**
11. Run the `GET /api/v1/inboundcontact/` endpoint
    * [x] It should show 2 results
12. Run the `PATCH /api/v1/profiles/{id}` endpoint to update your profile to `store_phone_log:false`
13. Send a text to your local Relay number
14. Run the `GET /api/v1/inboundcontact/` endpoint
    * [x] It should show only the 2 results from step 11
